version: "3.9"

services:
  portainer:
    image: "portainer/portainer-ce:2.19.4-alpine"
    container_name: "portainer"
    restart: unless-stopped
    logging:
      driver: "json-file"
      options:
        max-size: "200k"
        max-file: "10"
    networks:
      - net
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.portainer.rule=Host(`portainer.${HOST}`)"
      - "traefik.http.routers.portainer.entrypoints=websecure"
      - "traefik.http.routers.portainer.tls.certresolver=myresolver"
      - "traefik.http.services.portainer.loadbalancer.server.port=9000"
      - "traefik.http.middlewares.portainersts.headers.stsincludesubdomains=false"
      - "traefik.http.middlewares.portainersts.headers.stspreload=true"
      - "traefik.http.middlewares.portainersts.headers.stsseconds=31536000"
      - "traefik.http.middlewares.portainersts.headers.isdevelopment=false"
      - "traefik.http.routers.service.middlewares=portainersts"
    volumes:
      - "portainer_data:/data"
      - "/var/run/docker.sock:/var/run/docker.sock:ro"

networks:
  net:
    external:
      name: traefik_net

volumes:
  portainer_data:
